lib/sycamore/sycamore/docset.py

```diff
@@ -903,6 +906,28 @@ def map(self, f: Callable[[Document], Document], **resource_args) -> "DocSet":
         mapping = Map(self.plan, f=f, **resource_args)
         return DocSet(self.context, mapping)
 
+    def kmeans(self, K: int, iterations: int, init_mode: str = "random", epsilon: float = 1e-4):
+        """
+        Apply kmeans over embedding field
+
+        Args:
+            K: the count of centroids
+            iterations: the max iteration runs before converge
```

Comment:
Is there a reasonable default for this? I at least wouldn't know what a good value to pick would be.

Reply:
spark uses 20, we could follow the same, but it should really be a tuning process.